Kava/Kava/Source/PageObjects/Infrastructure/Elements/CollectionView.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  CollectionView.swift
+//  Kava
+//
+//  Created by John Hammerlund on 3/9/16.
+//  Copyright © 2016 MINDBODY, Inc. All rights reserved.
+//
+
+import Foundation
+import XCTest
+
+public enum ScrollDirection {
+    case Horizontal
+    case Vertical
+}
+
+/**
+ A collection view, which could be a table view. The provided generic argument 
+ allows us to infer the proceeding block, defaulted to scope of the application
+ */
+public class CollectionView<TResultBlock : Block> : Element {
+    
+    public typealias TResultBlockBuilder = (() -> TResultBlock)?
+    
+    private let scrollDirection: ScrollDirection
+    
+    public var visibleCells: [Swipeable<TResultBlock>] {
+        get {
+            return self.backingElement.cells.allElementsBoundByIndex.map {
+                return Swipeable(parentBlock: self.parentBlock, backingElement: $0)
```

Comment:
Syntactic sugar -- the block type is actually defined as ((element: Self.ElementType) -> TOut) where TOut is inferred from the closure return statement. $0, $1, $2, etc. is a shorthand for referring to the arguments in the closure. The above could be rewritten as such:
```swift
return self.backingElement.cells.allElementsBoundByIndex.map { (element: XCUIElement) -> 
    return Swipeable(parentBlock: self.parentBlock, backingElement: $0)
}
```